src/core/Solver_intf.ml

```diff
@@ -67,17 +62,24 @@ type 'term eval_res =
     - [Valued (true, [y])] if [y] is assigned to [0]
     - [Valued (false, [x; y])] if [x] and [y] are assigned to 1 (or any non-zero number)
 *)
+type 'term eval_res =
+  | Unknown  (** The given formula does not have an evaluation *)
+  | Valued of bool * 'term list
+      (** The given formula can be evaluated to the given bool.
+                                      The list of terms to give is the list of terms that
+                                      were effectively used for the evaluation. *)
 
+(** Asusmptions made by the core SAT solver. *)
 type ('term, 'formula, 'value) assumption =
   | Lit of 'formula  (** The given formula is asserted true by the solver *)
-  | Assign of 'term * 'value (** The term is assigned to the value *)
-(** Asusmptions made by the core SAT solver. *)
+  | Assign of 'term * 'value  (** The term is assigned to the value *)
 
+(** The type of reasons for propagations of a formula [f]. *)
 type ('term, 'formula, 'proof) reason =
   | Eval of 'term list
-  (** The formula can be evalutaed using the terms in the list *)
+      (** The formula can be evalutaed using the terms in the list *)
```

Comment:
Again with the weird arbitrary indentation of doc comments.
Note that ocp-indent indents these as they currently are.